sparkmagic/sparkmagic/livyclientlib/command.py

```diff
@@ -42,6 +44,14 @@ def execute(self, session):
             response = session.http_client.post_statement(session.id, data)
             statement_id = response[u'id']
             output = self._get_statement_output(session, statement_id)
+        except KeyboardInterrupt as e:
+            if statement_id >= 0:
+                response = session.http_client.cancel_statement(session.id, statement_id)
+                session.wait_for_idle()
```

Comment:
I noticed here that session will wait for idle here for the statement cancellation. Based on my previous experience with Livy, cancel statement can fail and will always fail for non-spark code due to Livy's implementation (they might have done works after my experiments). You might need to consider that into this MR

Reply:
Thanks for the suggestion. When the execution reaches here, all the user wants is to interrupt the execution. How about I just try to catch everything here? I can also pass a shorter timeout to `wait_for_idle()`

Reply:
Hey Hanyu. Yes I understand that interruption is all we want to achieve but as you mentioned, we might get timeout for idle waiting and what's worse is that Livy Session will become unusable after a "unsupported" cancel call. What I would suggest is that we catch that timeout exception and print helpful exception msg telling user like the session might be dead due to the cancellations and restart might be required.